Makefile

```diff
@@ -1,8 +1,47 @@
 all: build
 
-include csi.mk
+CSI_SPEC := spec.md
+CSI_PROTO := csi.proto
 
-build: $(CSI_PROTO) $(CSI_GOSRC)
+# This is the target for building the temporary CSI protobuf file.
+#
+# The temporary file is not versioned, and thus will always be
+# built on Travis-CI.
+$(CSI_PROTO).tmp: $(CSI_SPEC)
+	cat $? | \
+	  sed -n -e '/```protobuf$$/,/```$$/ p' | \
+	  sed -e 's@^```.*$$@////////@g' > $@
+
+# This is the target for building the CSI protobuf file.
+#
+# This target depends on its temp file, which is not versioned.
+# Therefore when built on Travis-CI the temp file will always
+# be built and trigger this target. On Travis-CI the temp file
+# is compared with the real file, and if they differ the build
+# will fail.
+#
+# Locally the temp file is simply copied over the real file.
+$(CSI_PROTO): $(CSI_PROTO).tmp
+ifeq (true,$(TRAVIS))
+	diff "$@" "$?"
+else
+	diff "$@" "$?" > /dev/null 2>&1 || cp -f "$?" "$@"
+.PHONY: $(CSI_PROTO).tmp
+endif
+
+build: $(CSI_PROTO)
+
+# If this is not running on Travis-CI then for sake of convenience
+# go ahead and update the language bindings as well.
+ifneq (true,$(TRAVIS))
```

Comment:
Why not build library as well in CI? We need to capture library build error on CI as well, right?